src/data/number/numberFn.ts

```diff
@@ -1,10 +1,90 @@
 import { assert, assertInteger } from '../../utils/assert'
-import { plString } from '../string/stringFn'
 import { PLNumber } from './PLNumber'
 
-export const plNumber = (value: number): PLNumber => new PLNumber(value)
+export interface DecimalResult {
+  intValue: number
+  decimals: number
+}
+
+export function plFloat(x: number): PLNumber {
+  return parseNumber(x.toString())
+}
+
+export function plNumber(intValue: number, decimals = 0): PLNumber {
+  assertInteger(intValue)
+  assertInteger(decimals)
+  return new PLNumber(intValue, decimals)
+}
+
+export function parseNumber(strValue: string): PLNumber {
+  assertNumeric(strValue)
+  const decimalObj = isScientific(strValue) ? parseScientificString(strValue) : parseDecimalString(strValue)
+  return plNumber(decimalObj.intValue, decimalObj.decimals)
+}
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export function assertNumeric(strValue: any): boolean {
+  return assert(strValue === '' || isNaN(strValue), `Invalid number: "${strValue}"`)
+}
+
+export const isRational = (x: number): boolean => {
+  const decimalStr = x.toString().split('.').pop()
+  const decimalLen = decimalStr ? decimalStr.length : 0
+  const isFraction = /\..*(\d)\1{12,}\d$/g.test(x.toString())
+  return decimalLen < 12 ? true : isFraction
```

Comment:
why `12`?
If you must use a contant value, please crate a `const` outside of the this function and git it a proper name

Reply:
I've removed this function.